src/ImageMagick.jl

```diff
@@ -228,11 +228,23 @@ mapIM(x::Normed) = x
 # Make the data contiguous in memory, this is necessary for
 # imagemagick since it doesn't handle stride.
 to_contiguous(A::Array) = A
-to_contiguous(A::AbstractArray) = convert(Array, A)
-to_contiguous(A::SubArray) = copy(A)
+to_contiguous(A::AbstractArray) = collect1(A)
 to_contiguous(A::BitArray) = convert(Array{N0f8}, A)
 to_contiguous(A::ColorView) = to_contiguous(channelview(A))
 
+if VERSION < v"0.6.0-pre.beta.28"
+    collect1(A) = collect_indices(indices(A), A)
+    collect_indices(::Tuple{}, A) = copy!(Array{eltype(A)}(), A)
+    collect_indices(indsA::Tuple{Vararg{OneTo}}, A) =
+        copy!(Array{eltype(A)}(length.(indsA)), A)
+    function collect_indices(indsA, A)
+        B = Array{eltype(A)}(length.(indsA))
+        copy!(B, CartesianRange(indices(B)), A, CartesianRange(indsA))
+    end
+else
+    const collect1 = collect
+end
+
```

Comment:
Probably worth CCing @tkelman here. I don't think https://github.com/JuliaLang/julia/pull/21257 can safely be backported to julia-0.5. I guess the question is whether Compat should define `collect1`?